utils/jwtUtil.js

```diff
@@ -56,25 +97,49 @@ function verifyToken(req, cb) {
   }
 
   if (token) {
-    jwt.verify(token, env.tokenSecret, {}, (err, decodedData) => {
+    jwt.verify(token, secret, {}, (err, decodedData) => {
       if (err) {
-        handleInvalidToken(cb);
-      } else {
-        User.findOne({ where: { name: decodedData.username } })
+        return handleInvalidToken(cb);
+      } else { // eslint-disable-line no-else-return
+        return User.findOne({ where: { name: decodedData.username } })
         .then((user) => {
-          // set user in request
           if (user) {
-            req.user = user;
-            return cb();
-          }
+            req.user = user; // set user in request
 
-          return handleInvalidToken(cb);
+            /*
+             * No need to check the token record if this is the default UI
+             * token.
+             */
+            if (decodedData.username === decodedData.tokenname) {
```

Comment:
With this code in place, we will not restrict a user who created token (with tokenName =username) and the token is revoked. We can merge this code for now. What we can do in another story is that we can keep the existing tokens as it is, add a field `isUIToken: true` to the tokens created on the fly from UI, and then add a check `if (decodedData.isUIToken && decodedData.isUIToken ==true) { cb() }` instead of check `if (decodedData.username === decodedData.tokenname) { cb() }`
Let me create a story for that.